### DISCUSSION TITLE
Make send size limit configurable

### DISCUSSION THREAD

**Discussion by parisni** [dani-garcia/vaultwarden#1804] [Ideas]
The limit for `Send` is 100mo and apparently there is no env variable to configure it.

**Comment by BlackDex** [comment]
This limit is a hard limit also configured by Bitwarden on all aspects. All clients have a limit of 100MB. So the only thing we could do is lower the amount, which i think is not that useful maybe.

**Comment by lennvilardi** [comment]
is this still true ? Limit is now seems set to 500Mb can we go further and how ?

**Reply by BlackDex** [reply]
The limit is still hardcoded within the clients: https://github.com/bitwarden/jslib/blob/60878cd4ed9b1989860a918b62ce72529232560f/angular/src/components/attachments.component.ts#L67-L68=

**Reply by dillfrescott** [reply]
Awww 500 MB isnt enough for me :(

**Reply by BlackDex** [reply]
Thats all you can get

**Reply by dillfrescott** [reply]
Yea, sadly

**Reply by tessus** [reply]
I am actually glad that people don't use my vaultwarden server for sending 4k Blu-ray images.

**Reply by bitnapper** [reply]
Well, I wanted to send a dashcam video from an accident when a car pushed me with my bike on purpose from the street to the police via bitwarden because I trust my own bitwarden instance more then most filesharing services. Also the police has a limit of 100mb for their online tool and 10mb limit on email.

Since it is my own server I don't see any technical reasons for this hard limit besides. A default limit is ok but why shouldn't one set a higher limit?

Also @tessus, just making fun of people who may have legitimate reasons for wanting their on instances to have a configurable limit is pretty condescending.  You can have any limit you want an your server but that you don't have a legitimate application for higher limits doesn't mean that every others is illegitemate.

**Reply by BlackDex** [reply]
Doesn't matter what we want, the clients just do not handle it. Mainly because of encryption in memory etc.. 
See the following code lines which shows the limit is 500MB.
- https://github.com/bitwarden/clients/blob/dedd7f1b5cc66038abb120b4fae83666c487ad04/libs/angular/src/tools/send/add-edit.component.ts#L315
- https://github.com/bitwarden/clients/blob/dedd7f1b5cc66038abb120b4fae83666c487ad04/libs/angular/src/vault/components/attachments.component.ts#L70
- https://github.com/bitwarden/clients/blob/dedd7f1b5cc66038abb120b4fae83666c487ad04/libs/vault/src/cipher-form/components/attachments/cipher-attachments.component.ts#L174
- https://github.com/bitwarden/clients/blob/dedd7f1b5cc66038abb120b4fae83666c487ad04/libs/vault/src/cipher-form/components/attachments/cipher-attachments.component.spec.ts#L208

**Reply by tessus** [reply]
@bitnapper I think you misunderstood. I wasn't making fun of anyone. I adhere in my corrspondence to The Four Agreements and Crocker's rules as much as possible.

I was stating (because it's a client issue (thus any discussion is moot anyway) and there's no server side way to limit the send size, unless you count the overall send size limit per user) that I am glad that my users cannot use the sends to share Blurays or such.
I'm not sure what this has to do with you or anyone else.
